lib/srv/desktop/windows_server.go

```diff
@@ -660,7 +752,45 @@ func (s *WindowsService) connectRDP(ctx context.Context, log logrus.FieldLogger,
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
+	delay := timer()
 	tdpConn := tdp.NewConn(proxyConn)
+	sessionStartTime := s.cfg.Clock.Now().UTC().Round(time.Millisecond)
+	tdpConn.OnSend = func(m tdp.Message, b []byte) {
```

Comment:
Maybe give the event handlers names so there is less noise inside connectRDP? That way you could also test them separately.